src/esw/brushed_motors/ROSHandler.cpp

```diff
@@ -106,3 +155,70 @@ void ROSHandler::moveMastGimbal(const mrover::MastGimbal::ConstPtr& msg) {
     moveControllerOpenLoop("mast_gimbal_up_down", (float) msg->up_down);
     moveControllerOpenLoop("mast_gimbal_left_right", (float) msg->left_right);
 }
+
+// REQUIRES: valid req and res objects
+// MODIFIES: res
+// EFFECTS: sends a move/calibration command to the mcu
+bool ROSHandler::processMotorCalibrate(mrover::CalibrateMotors::Request& req, mrover::CalibrateMotors::Response& res) {
+    auto controller_iter = ControllerMap::controllersByName.find(req.name);
+    auto& [name, controller] = *controller_iter;
+
+    // Check if already calibrated
+    controller->askIsCalibrated();
+
+    // Determine if calibration is needed
+    bool shouldCalibrate = !(controller_iter == ControllerMap::controllersByName.end()
+                             || controller->getCalibrationStatus()
+                             || !controller->getLimitSwitchEnabled());
+
+    // Calibrate
+    if (shouldCalibrate) {
+        controller->moveOpenLoop(controller->calibrationVel);
+        res.actively_calibrating = true;
+    } else {
+        res.actively_calibrating = false;
+    }
+
+    return true;
+}
+
+// REQUIRES: valid req and res objects
+// MODIFIES: res
+// EFFECTS: hard sets the requested controller angle
+bool ROSHandler::processMotorAdjust(mrover::AdjustMotors::Request& req, mrover::AdjustMotors::Response& res) {
+
+    auto controller_iter = ControllerMap::controllersByName.find(req.name);
+
+    if (controller_iter == ControllerMap::controllersByName.end()) {
+        ROS_ERROR("Could not find controller named %s.", req.name.c_str());
+        res.success = false;
+        return false;
+    }
+
+    auto& [name, controller] = *controller_iter;
+    controller->overrideCurrentAngle(req.value);
```

Comment:
I commented above we need to set calibrated to true in this function, I suppose it could go here or in that function?

Reply:
Mentioned above that it is covered by nucleo code